jmclient/jmclient/configure.py

```diff
@@ -612,10 +593,26 @@ def get_mchannels(mode="TAKER"):
             onion_data["serving"] = False
         onion_data['btcnet'] = get_network()
         # Just to allow a dynamic set of var:
-        onion_data["section-name"] = section
-        configs.append(onion_data)
+        onion_data["section-name"] = s
+        return onion_data
 
-    return configs
+    # processing the IRC sections:
+    onion_sections = []
+    irc_sections = []
+    for section in jm_single().config.sections():
+        if not section.startswith(SECTION_NAME):
+            continue
+        if jm_single().config.has_option(section, "type"):
+            channel_type = jm_single().config.get(section, "type").lower()
+            if channel_type == "onion":
+                onion_sections.append(get_onion_section(section))
+            elif channel_type == "irc":
+                irc_sections.append(get_irc_section(section))
+            else:
+                irc_sections.append(get_irc_section(section))
+        assert irc_sections or onion_sections
+        assert len(onion_sections) < 2
```

Comment:
The two asserts should be outside of the loop.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1182/commits/6a4cf97887a1abfd172362cced22636527bead41